### PR TITLE
PHP 8: Code Review `Poll`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @smeyer-ilias / @tfamula,

I completed the review of the `Modules/Poll` component.

Results:
- [x] The code style is `PSR-2 + X` compliant: 1 File has been changed by the `PHP-CS-FIXER`
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [x] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

I also fixed some trivial issues reported by my inspection profile.

--

Actions needed:

- [ ] Please check my inline comments. Some of them are just information, some require an action. You can use `TODO PHP8-REVIEW` when searching.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Modules/Poll/classes/class.ilPollUserTableGUI.php [hunk 1]
```diff
@@ -105,9 +105,9 @@ protected function fillRowCSV(ilCSVWriter $a_csv, array $a_set) : void
         $a_csv->addColumn((string) ($a_set["firstname"] ?? ''));
         foreach ($this->answer_ids as $answer_id) {
             if ($a_set["answer" . $answer_id]) {
-                $a_csv->addColumn(true);
+                $a_csv->addColumn('true');// TODO PHP8-REVIEW I am not sure if my fix is correct
```
Comment (mjansenDatabay) [inline Modules/Poll/classes/class.ilPollUserTableGUI.php:108]: I am not sure if my fix is correct

File: Modules/Poll/classes/class.ilPollUserTableGUI.php [hunk 2]
```diff
@@ -105,9 +105,9 @@ protected function fillRowCSV(ilCSVWriter $a_csv, array $a_set) : void
...
             } else {
-                $a_csv->addColumn(false);
+                $a_csv->addColumn('false');// TODO PHP8-REVIEW I am not sure if my fix is correct
```
Comment (mjansenDatabay) [inline Modules/Poll/classes/class.ilPollUserTableGUI.php:110]: I am not sure if my fix is correct

File: Modules/Poll/classes/class.ilPollUserTableGUI.php (hunk 1 above)
Comment (smeyer-ilias) [reply Modules/Poll/classes/class.ilPollUserTableGUI.php:108]: In addColumn() the csv-string is concatenated with the .= operator and thus an implicit string cast is performed. 
So I guess addColumn('1'); should be compatible with previous implementation.

File: Modules/Poll/classes/class.ilPollUserTableGUI.php (hunk 2 above)
Comment (smeyer-ilias) [reply Modules/Poll/classes/class.ilPollUserTableGUI.php:110]: In addColumn() the csv-string is concatenated with the .= operator and thus an implicit string cast is performed.
So I guess addColumn(''); should be compatible with previous implementation.

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]
Thanks, for reviewing and fixing issues.

**Comment by smeyer-ilias** [comment]
I merge it and will fix the csv->addColumn issue.

**Pull request merged**